### PR TITLE
Config for HUD rendering

### PR DESCRIPTION
**Pull request by ZZZank**
This PR:
- adds config for default values of the rendering of certain aspects of the HUD, e.g. Vignette.

## TL;DR

There're 14 new configs, each matching one flag in `net.minecraftforge.client.GuiIngameForge`. 

All new configs are in `client` scope of `config/forge.cfg`.

They will be applied on config changed, but mods are still allowed to modify flags in `GuiIngameForge` after that, to keep compatibility. 

I also added lang entries for them. ~~might not be accurate enough for illustrating the usage of these new options~~

### config screen

image
### example(vignette option)
|on|off|
|-|-|
| on| off|

### CODE REVIEW

**Comment by Ecdcaeb** [comment]
I do not think the `RequiresMcRestart` is nessary.
On the one hand, it's just a switch, on the other hand, [Forge will reload it](https://github.com/CleanroomMC/Cleanroom/blob/49b9029224e5ec9dad07a29f74398c37c1545e03/src/main/java/net/minecraftforge/common/ForgeModContainer.java#L489). We should sync it when it changed. (`load == false`).

Maybe we can add a new subcategory called hub in client.

In my testing it worked just fine, it's a nice and meaningful pr.^~^

**Comment by ZZZank** [comment]
> [Forge will reload it](https://github.com/CleanroomMC/Cleanroom/blob/49b9029224e5ec9dad07a29f74398c37c1545e03/src/main/java/net/minecraftforge/common/ForgeModContainer.java#L489). We should sync it when changed.

I didn't know about that before🤔surely it will be a nice addition, I'll take a look.
> Maybe we can add a new subcategory called `hub` in `client`. (`client.hub`)

Good idea, will take a look too.

**Comment by ZZZank** [comment]
Should be good to go now.
- onConfigChanged() method is slightly refactored, to make client config reloading clearer. This was intended for implementing config refresh of `GuiIngameForge`, but it's now replaced by:
- Addd configs are now directly linked to flags in `GuiIngameForge`, so new class members and refreshing are no longer needed. 
- Sub category is NOT implemented, because config options will "leak" into its parent category. 

an example of "options leaking", those 3 configs are originally placed in `client/hud` sub category, but somehow get duplicated and show up in `client` category:
image

**Comment by ZZZank** [comment]
Update: Thanks @Ecdcaeb for pointing out that "leaked options" are actually just old options, preserved but unused, hence not taking effect.

Now new options are placed in a sub category called `HUD Settings`. 
image

File: src/main/java/net/minecraftforge/client/GuiIngameForge.java [hunk 1]
```diff
@@ -83,6 +87,9 @@ public class GuiIngameForge extends GuiIngame
     public static boolean renderFood = true;
     public static boolean renderHealthMount = true;
     public static boolean renderAir = true;
+    /**
+     * experiAnce?
```
Comment (Rongmario) [inline src/main/java/net/minecraftforge/client/GuiIngameForge.java:91]: Unneeded javadoc

File: src/main/java/net/minecraftforge/common/ForgeModContainer.java [hunk 2]
```diff
@@ -373,6 +370,14 @@ private static void syncConfig(boolean load)
         prop.setLanguageKey("forge.configgui.selectiveResourceReloadEnabled");
         propOrder.add(prop.getName());
 
+        var categoryHudId = CATEGORY_CLIENT + Configuration.CATEGORY_SPLITTER + "hud";
+        var categoryHud = config.getCategory(categoryHudId);
+        categoryHud.setComment("Controls the rendering of certain aspects of the HUD.");
```
Comment (Rongmario) [inline src/main/java/net/minecraftforge/common/ForgeModContainer.java:375]: `Controls rendering of various HUD elements`

File: src/main/java/net/minecraftforge/common/ForgeModContainer.java [hunk 3]
```diff
@@ -381,32 +386,88 @@ private static void syncConfig(boolean load)
...
+        };
+
+        var order = new ArrayList<String>(allConfigs.length);
+        var values = new boolean[allConfigs.length];
+
+        for (int i = 0; i < values.length; i++)
+        {
+            var key = allConfigs[i][0];
+            var friendlyName = allConfigs[i][1];
+            var comment = "Toggle off to force disable %s rendering. Note that this config is applied on config saved, mods can still change %s rendering behaviour after that."
```
Comment (Rongmario) [inline src/main/java/net/minecraftforge/common/ForgeModContainer.java:419]: `Toggle off to disable rendering of the %s. Note that even though this config is applied when the config is saved to disk, mods can still override the behaviour after-the-fact.`

File: src/main/resources/assets/forge/lang/en_US.lang [hunk 4]
```diff
@@ -92,6 +92,38 @@ forge.configgui.allowEmissiveItems.tooltip=Allow item rendering to detect emissi
 forge.configgui.selectiveResourceReloadEnabled=Enable Selective Resource Loading
 forge.configgui.selectiveResourceReloadEnabled.tooltip=When enabled, makes specific reload tasks such as language changing quicker to run.
 
+forge.configgui.hud.category=HUD Settings
+forge.configgui.hud.category.tooltip=Controls the rendering of certain aspects of the HUD.
```
Comment (Rongmario) [inline src/main/resources/assets/forge/lang/en_US.lang:96]: `Controls rendering of various HUD elements`

File: src/main/resources/assets/forge/lang/en_US.lang [hunk 5]
```diff
@@ -92,6 +92,38 @@ forge.configgui.allowEmissiveItems.tooltip=Allow item rendering to detect emissi
...
+forge.configgui.hud.allowAir=Allow Air Rendering
+forge.configgui.hud.allowAir.tooltip=Toggle off to force disable air rendering. Note that this config is applied on config saved, mods can still change air rendering behaviour after that.
+forge.configgui.hud.allowExperience=Allow Experience Rendering
+forge.configgui.hud.allowExperience.tooltip=Toggle off to force disable experience rendering. Note that this config is applied on config saved, mods can still change experience rendering behaviour after that.
+forge.configgui.hud.allowJumpBar=Allow Jump Bar Rendering
+forge.configgui.hud.allowJumpBar.tooltip=Toggle off to force disable jump bar rendering. Note that this config is applied on config saved, mods can still change jump bar rendering behaviour after that.
+forge.configgui.hud.allowObjective=Allow Objective Rendering
+forge.configgui.hud.allowObjective.tooltip=Toggle off to force disable objective rendering. Note that this config is applied on config saved, mods can still change objective rendering behaviour after that.
```
Comment (Rongmario) [inline src/main/resources/assets/forge/lang/en_US.lang:125]: Align it with the aforementioned `Toggle off to disable rendering of the %s. Note that even though this config is applied when the config is saved to disk, mods can still override the behaviour after-the-fact.`

**Review by Rongmario — CHANGES_REQUESTED** [review]

File: src/main/java/net/minecraftforge/client/GuiIngameForge.java (hunk 1 above)
Comment (ZZZank) [reply src/main/java/net/minecraftforge/client/GuiIngameForge.java]: It's intended for making devs aware that this field actually refers to `experience`. I've added some more doc to it, should be better now.

**Review by Rongmario — APPROVED** [review]

**Pull request merged**